examples/cloud_discovery_service/library_api/c/cloud_discovery_service_library_api.c

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ (c) 2005-2023 Copyright, Real-Time Innovations, Inc. All rights reserved.
+ RTI grants Licensee a license to use, modify, compile, and create derivative
+ works of the Software. Licensee has the right to distribute object form only
+ for use with RTI products. The Software is provided "as is", with no warranty
+ of any type, including any warranty for fitness for any purpose. RTI is under
+ no obligation to maintain or support the Software. RTI shall not be liable for
+ any incidental or consequential damages arising out of the use or inability to
+ use the software.
+ ******************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "ndds/ndds_c.h"
+#include "clouddiscoveryservice/clouddiscoveryservice_service.h"
+
+static int service_main(const char *cfgName, const char *cfgFile)
+{
+    struct RTI_CDS_Service *service = NULL;
+    struct RTI_CDS_Property property = RTI_CDS_Property_INITIALIZER;
+    DDS_Boolean ok = DDS_BOOLEAN_FALSE;
+    struct DDS_Duration_t sleepPeriod = { 0, 1000000000 };
+
+    property.cfg_file = DDS_String_dup(cfgFile);
+    property.service_name = DDS_String_dup(cfgName);
+    property.application_name =
+            DDS_String_dup("CloudDiscoveryService-LibraryAPI-C");
+
+    /* Dial up the logging at service level */
+    property.service_verbosity = RTI_CDS_LOG_VERBOSITY_ALL;
+    property.dds_verbosity = RTI_CDS_LOG_VERBOSITY_EXCEPTIONS;
+
+
+    service = RTI_CDS_Service_new(&property);
+    if (service == NULL) {
+        printf("Error creating the Cloud Discovery Service instance\n");
+        return -1;
+    }
+
+    ok = RTI_CDS_Service_start(service);
+    if (!ok) {
+        printf("Error starting the Cloud Discovery Service instance\n");
+        return -1;
+    }
+
+    printf("ConfigName=%s, ConfigFile=%s\n", cfgName, cfgFile);
+    /* Infinite loop until the user terminates the process */
+    while (1) {
+        NDDS_Utility_sleep(&sleepPeriod);
+    }
+
+    ok = RTI_CDS_Service_stop(service);
+    if (!ok) {
+        printf("Error stopping the Cloud Discovery Service instance\n");
+        return -1;
+    }
+
+    /* Sample usage of deleting the service instance for resource cleanup */
+    /* RTI_CDS_Service_delete(service); */
+
+    /* Sample usage of deleting the service property to free the strings */
+    /* RTI_CDS_Property_finalize(&property); */
```

Comment:
We're missing a return,
```suggestion
    /* RTI_CDS_Property_finalize(&property); */
    
    return EXIT_SUCCESS;
```

Reply:
Good catch. Done.